apps/backend/src/repositories/memo-repository.ts

```diff
@@ -0,0 +1,369 @@
+    sourceMessageId: row.source_message_id,
+    sourceConversationId: row.source_conversation_id,
+    title: row.title,
+    abstract: row.abstract,
+    keyPoints: row.key_points,
+    sourceMessageIds: row.source_message_ids,
+    participantIds: row.participant_ids,
+    knowledgeType: row.knowledge_type as KnowledgeType,
+    tags: row.tags,
+    parentMemoId: row.parent_memo_id,
+    status: row.status as MemoStatus,
+    version: row.version,
+    revisionReason: row.revision_reason,
+    createdAt: row.created_at,
+    updatedAt: row.updated_at,
+    archivedAt: row.archived_at,
+  }
+}
+
+const SELECT_FIELDS = `
+  id, workspace_id, memo_type, source_message_id, source_conversation_id,
+  title, abstract, key_points, source_message_ids, participant_ids,
+  knowledge_type, tags, parent_memo_id, status, version, revision_reason,
+  created_at, updated_at, archived_at
+`
+
+const SELECT_FIELDS_PREFIXED = `
+  m.id, m.workspace_id, m.memo_type, m.source_message_id, m.source_conversation_id,
+  m.title, m.abstract, m.key_points, m.source_message_ids, m.participant_ids,
+  m.knowledge_type, m.tags, m.parent_memo_id, m.status, m.version, m.revision_reason,
+  m.created_at, m.updated_at, m.archived_at
+`
+
+export const MemoRepository = {
+  async findById(client: PoolClient, id: string): Promise<Memo | null> {
+    const result = await client.query<MemoRow>(sql`SELECT ${sql.raw(SELECT_FIELDS)} FROM memos WHERE id = ${id}`)
+    if (!result.rows[0]) return null
+    return mapRowToMemo(result.rows[0])
+  },
+
+  async findByWorkspace(
+    client: PoolClient,
+    workspaceId: string,
+    options?: { status?: MemoStatus; type?: MemoType; limit?: number }
+  ): Promise<Memo[]> {
+    const limit = options?.limit ?? 50
+    const conditions: string[] = [`workspace_id = $1`]
+    const values: unknown[] = [workspaceId]
+    let paramIndex = 2
+
+    if (options?.status) {
+      conditions.push(`status = $${paramIndex++}`)
+      values.push(options.status)
+    }
+    if (options?.type) {
+      conditions.push(`memo_type = $${paramIndex++}`)
+      values.push(options.type)
+    }
+
+    values.push(limit)
+
+    const query = `
+      SELECT ${SELECT_FIELDS} FROM memos
+      WHERE ${conditions.join(" AND ")}
+      ORDER BY created_at DESC
+      LIMIT $${paramIndex}
+    `
+
+    const result = await client.query<MemoRow>(query, values)
+    return result.rows.map(mapRowToMemo)
+  },
+
+  async findByStream(
+    client: PoolClient,
+    streamId: string,
+    options?: { status?: MemoStatus; limit?: number; orderBy?: "createdAt" | "updatedAt" }
+  ): Promise<Memo[]> {
+    const limit = options?.limit ?? 50
+    const orderBy = options?.orderBy === "updatedAt" ? "updated_at" : "created_at"
+
+    // Use UNION to fetch both:
+    // 1. Conversation memos (via source_conversation_id -> conversations.stream_id)
+    // 2. Message memos (via source_message_id -> messages.stream_id)
+    if (options?.status) {
```

Comment:
Perhaps we can do better with the querying so that we don't have nearly the same query twice?

Reply:
Refactored to use a single query with an optional status clause using `sql.raw()`. The status is now composed once and applied to both UNION branches.